### DISCUSSION TITLE
Jira Integration from GitLab pipeline

### DISCUSSION THREAD

**Discussion by kayvonkhosrowpour** [checkmarx-ltd/cx-flow#751] [Q&A]
## Background
I have cx-flow integrated with my GitLab pipeline by following the [GitLab integration docs](https://checkmarx.atlassian.net/wiki/spaces/SD/pages/1929937052/GitLab+Integration). 

I have the following already implemented:
1. Run an incremental scan on merge requests and add the comment to the GitLab merge request ✅
2. Run full scans on the main/master branch ✅

## Question
I would like to _also_ automatically create/update Issues in Jira within GitLab pipeline integration❓. Is this possible from a GitLab job?

**Comment by jbrotsos** [comment]
@kayvonkhosrowpour 

The workflow is usually this....

1. On Merge Request, a Checkmarx scan is done and the Merge Request comment is decorated with the scan results
2. The Merge Request is submitted and JIRA issues are created.

You can do this by overriding the variable in the GL template named "CX_FLOW_BUG_TRACKER".  The value should be "Jira" instead of "GitLab"

To do the configuration, I recommend using config-as-code and override the variable named "PARAMS" to have --config=cx.config --jira.url=${{secrets.JIRA_URL}} --jira.username=${{secrets.JIRA_USER}} --jira.token=${{secrets.JIRA_TOKEN}

Look at https://github.com/checkmarx-ltd/cx-flow/wiki/Tutorials#github-actions-with-jira-integration for tutorial help on the config-as-code.

**Reply by kayvonkhosrowpour** [reply]
@jbrotsos 

Thank you for pointing me to the right place. I assumed because the docs referenced GitHub it did not necessarily support GitLab. I will try this. However, I only have access to non-cloud hosted Jira. As such, it appears that only basic authentication (username + password) is supported: [https://docs.gitlab.com/ee/user/project/integrations/jira.html](https://docs.gitlab.com/ee/user/project/integrations/jira.html).  Is there support for this or is the token required?
